src/main/kotlin/org/rust/lang/core/resolve/NameResolution.kt

```diff
@@ -826,7 +827,9 @@ fun processMacroCallVariantsInScope(context: PsiElement, processor: RsResolvePro
     val result = MacroResolver.processMacrosInLexicalOrderUpward(context, processor)
     if (result) return true
 
-    val prelude = context.contextOrSelf<RsElement>()?.findDependencyCrateRoot(STD) ?: return false
+    val element = context.contextOrSelf<RsElement>() ?: return false
+    val crateRoot = element.crateRoot as? RsFile ?: return false
+    val prelude = implicitStdlibCrate(crateRoot)?.crateRoot ?: return false
```

Comment:
Looks like we don't need `element.crateRoot` here. You can pass any `RsFile` to `implicitStdlibCrate` directly. So I think we need something like 
```suggestion
    val file = context.contextualFile as? RsFile ?: return false
    val prelude = implicitStdlibCrate(file)?.crateRoot ?: return false
```

Reply:
Hmm, I'm not sure if that would work. If `lib.rs` contains `#![no_std]`, we want to stop resolving macros and `std` in all files imported by `lib.rs`, right? Using `contextualFile` would only stop `std` macro resolving if the file where the macro is used has `#![no_std]`.

Reply:
Yep. You are right.
Interesting fact that `implicitStdlibCrate` was written by me :sweat_smile: